CAIPs/caip-25.md

```diff
@@ -12,77 +12,45 @@ requires: 2, 10, 171, 217
-
-Conversely, a succesful response will contain all the required properties *and
-the provider's choice of the optional properties* expressed in a single unified
-`scopeObject`. In the case of identically-keyed `scopeObject`s appearing in both
-arrays in the request where properties from both are returned as authorized, the
-two scopes MUST be merged in the response (see examples below). However,
-respondents MUST NOT restructure scopes (e.g., by folding properties from a
-[CAIP-2][]-keyed, chain-specific scope object into a [CAIP-104][]-keyed,
-namespace-wide scope object) as this may introduce ambiguities (See Security
-Considerations below).
+The session is proposed by a caller and the response by the respondent is used as the baseline for an ongoing session that both parties will persist.
+The properties and authorization scopes that make up the session are expected to be persisted and tracked over time by both parties in a discrete data store, identified by an entropic [identifier][CAIP-171] assigned in the initial response.
+This object gets updated, extended, closed, etc. by successive calls and notifications, each tagged by this identifier.
+
+If a respondent (e.g. a wallet) needs to initiate a new session, whether due to user input, security policy, or session expiry reasons, it can simply generate a new session identifier to signal this notification to the calling provider; if a caller needs to initiate a new session, it can do so by sending a new request without a `sessionIdentifier`. In such cases, a respondent (e.g. wallet) may choose to explicitly close all sessions upon generation of a new one from the same origin or identity, or leave it to time-out; maintaining concurrent sessions is discouraged (see Security Considerations).
+
+Initial and ongoing authorization requests are grouped into two top-level arrays of [scopeObjects][CAIP-217], named `requiredScopes` and `optionalScopes`
+respectively.
+Each `scopeObject` in either array MUST be keyed uniquely within its parent, but these keys CAN appear in both
+(i.e., additional properties of an authorization target in `requiredScopes` may be requested in a separate `scopeObject` with the same key in the `optionalScopes` array).
+
+Each `scopeObject` in these arrays can be keyed to a specific [CAIP-2][] network identifier, or to an entire [CAIP-104][] namespace.
+`scopeObjects` keyed to an entire [CAIP-104][] namespace SHOULD contain a non-empty `chains` array to be actionable, making them functionally equivalent to a series of identical `scopeObjects`, each keyed to one of the members of `chains` expressed as a [CAIP-2][] scope.
+An empty or absent `chains` array SHOULD NOT be interpreted as a namespace-wide authorization (i.e. authorization for ANY network therein), but rather as a null authorization of 0 specied chains within that namespace.
+(See [CAIP-217][] for more details on the structure of the typed objects included in these arrays.)
```

Comment:
does the CAIP-217 spec also need to be updated then?

Reply:
not a bad idea, lemme look

Reply:
wow yeah they were out of sync in other ways too, embarassing!